discovery-infra/start_discovery.py

```diff
@@ -169,6 +170,25 @@ def _create_node_details(cluster_name):
     }
 
 
+def validate_dns(client, cluster_id):
+    if not args.managed_dns_domains:
+        # 'set_dns' (using dnsmasq) is invoked after nodes_flow
+        return
+
+    cluster = client.cluster_get(cluster_id)
+    api_address = "api.{}.{}".format(cluster.name, cluster.base_dns_domain)
+    ingress_address = "ingress.apps.{}.{}".format(cluster.name, cluster.base_dns_domain)
+    try:
+        api_vip = socket.gethostbyname(api_address)
```

Comment:
mm do u think it's enough? I guess even libvirt will return same results, no?
Maybe there is a python package that can tell u who did the resoltion? (instead of parsing dig)

Reply:
> mm do u think it's enough? I guess even libvirt will return same results, no?
> Maybe there is a python package that can tell u who did the resoltion? (instead of parsing dig)

Not sure it's crucial as set_dns is invoked only after this flow, so this indication should be good enough in this point. 
Any way, I'll try getting dns authority using [http://www.dnspython.org/](dnspython) in a separate PR to not block this.